clang/lib/Driver/Driver.cpp

```diff
@@ -6594,8 +6595,11 @@ bool clang::driver::isStaticArchiveFile(const StringRef &FileName) {
     // Any file with no extension should not be considered an Archive.
     return false;
   StringRef Ext(llvm::sys::path::extension(FileName).drop_front());
-  // Only .lib and .a files are to be considered.
-  return (Ext == "lib" || Ext == "a");
+  llvm::file_magic Magic;
+  llvm::identify_magic(FileName, Magic);
+  // Only .lib, .a and .lo files are to be considered.
+  return (Ext == "lib" ||
+          ((Ext == "a" || Ext == "lo") && Magic == llvm::file_magic::archive));
```

Comment:
Just curious, would we still need to check the extension after this? Or is `file_magic` not reliable to such an extent?
```suggestion
  return (Ext == "lib" || Magic == llvm::file_magic::archive));
```
Also, is there something `file_magic`-ish to cover Windows libraries? I wasn't able to find such a thing myself, though...

Reply:
`file_magic` can probably used explicitly without checking for extension.  It would open up the possibilities of identifying archives that aren't named as expected.

As for Windows, I didn't see anything that could be used for libraries: https://github.com/intel/llvm/blob/sycl/llvm/include/llvm/BinaryFormat/Magic.h

Reply:
Done.